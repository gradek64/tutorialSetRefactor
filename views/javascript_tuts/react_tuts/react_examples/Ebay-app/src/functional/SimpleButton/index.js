import React from 'react';
import PropTypes from 'prop-types';

// custom
import './simpleButton.scss';


const SimpleButton = ({ label, eventHandler }) => (
  <input
    className="simple-button"
    type="button"
    onClick={eventHandler}
    value={label}
  />
);
export default SimpleButton;

SimpleButton.propTypes = {
  eventHandler: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
};
