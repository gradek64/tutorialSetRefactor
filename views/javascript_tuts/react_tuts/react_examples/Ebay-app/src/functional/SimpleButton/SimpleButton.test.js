import React from 'react';
import { shallow } from 'enzyme';

import SimpleButton from './index';

test('renders correctly', () => {
  const wrapper = shallow(<SimpleButton label="label" eventHandler={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('button clicks and executed pass action method', () => {
  const actionCalled = jest.fn();
  const wrapper = shallow(<SimpleButton label="label" eventHandler={() => (actionCalled())} />);
  wrapper.find('input').simulate('click');

  // fucntion is called after click;
  expect(actionCalled).toHaveBeenCalledTimes(1);
});
