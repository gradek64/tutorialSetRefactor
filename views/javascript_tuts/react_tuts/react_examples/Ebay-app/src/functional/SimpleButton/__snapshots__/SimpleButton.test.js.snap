// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="simple-button"
  onClick={[Function]}
  type="button"
  value="label"
/>
`;
